### ISSUE TITLE
docker Ansible module does not start properly docker data containers

### ISSUE THREAD

**Issue by kpatryk** [ansible/ansible-modules-core#1550]
Hi,

Trying to start Oracle Data Container from Ansible docker core module. Previously there was no container of such name created in the system. After a while waiting the following error is thrown by Ansible:

`failed: [dev] => {"changed": false, "failed": true}
msg: Docker API Error: Conflict. The name "oracledata" is already in use by container ad653e28b23f. You have to delete (or rename) that container to be able to reuse that name.`

The mentioned container has been created by Ansible docker module during first playbook run that has failed. 

Ansible snippet code: 

```
  name: make sure Oracle Data container exists
  docker:
    name: oracledata
    image: "{{ test_registry_host if test_registry_enabled else registry_snapshot_host }}:{{ test_registry_port if test_registry_enabled else registry_snapshot_port }}/oracledata"
    state: present
    command: /bin/true
    insecure_registry: yes
    volumes:
    - "{{ oracle_tablespace_path }}"
  register: status_oracledata
  tags:
    - oracle
    - db-up
```

Remarks:
- Docker Registry 2.0 has been used to store locally images. Ansible pulls image from the private registry (no SSL configured). After task failure I noticed that still process of copying files to the file system has not been completed (image SHA was created under /var/lib/docker/vfs/dir/ directory). Docker version in provisioned machine: 

```
[vagrant@dev ~]$ docker version 
Client version: 1.6.2.el7
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): c3ca5bb/1.6.2
OS/Arch (client): linux/amd64
Server version: 1.6.2.el7
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): c3ca5bb/1.6.2
OS/Arch (server): linux/amd64
```

and docker info:

```
[vagrant@dev ~]$ docker info 
Containers: 2
Images: 33
Storage Driver: devicemapper
 Pool Name: docker-253:1-68054140-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 16.62 GB
 Data Space Total: 107.4 GB
 Data Space Available: 36.41 GB
 Metadata Space Used: 10.08 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.137 GB
 Udev Sync Supported: true
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.93-RHEL7 (2015-01-28)
Execution Driver: native-0.2
Kernel Version: 3.10.0-229.4.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
CPUs: 2
Total Memory: 2.782 GiB
Name: dev
ID: KEFU:NNQN:YEJF:2C2H:64VH:FYTF:C3OI:62T3:UZZ4:7QUQ:OQ2N:3EVW
```
- Second run has been successful. 
- Ansible version in the local machine: 

```
ansible 1.9.1
  configured module search path = None
```
- Python Docker installed dependencies in remote machine:

```
docker-compose (1.3.0rc2)
docker-py (1.2.2)
dockerpty (0.3.4)
```

**Comment by anthonygreen** [comment]
I'm seeing this as well; with the task: 

```
- name: Jenkins data directory
  docker: 
    name: jenkins.data 
    image: jenkins
    state: present
    volumes: "/home/{{ansible_ssh_user}}/.jenkins:/var/jenkins_home" 
  sudo: yes  
```

the container has been created but the task fails

```
failed: [ubuntu] => {"changed": false, "failed": true}
msg: Docker API Error: Conflict. The name "jenkins.data" is already in use by container cf07b63dd11e. You have to delete (or rename) that container to be able to reuse that name.

FATAL: all hosts have already failed -- aborting
```

**Comment by wayt** [comment]
The same here, in the tasks/main.yml : 

```
- name: api nginx-lb container
  docker:
    name: api-nginx-lb
    image: nginx
    state: reloaded
    #pull: always
    restart_policy: always
    count: 1
    volumes:
    - /home/api-nginx-lb/nginx.conf:/etc/nginx/nginx.conf:ro
    - /home/api-nginx-lb/sites-enabled:/etc/nginx/sites-enabled
    - /home/api-nginx-lb/ssl:/etc/nginx/ssl
    ports:
    - "80:80"
    - "443:443"
```

When I run the playbook:

```
TASK: [api-lb | api nginx-lb container] ***************************************
failed: [eu-api-lb-1] => {"changed": true, "failed": true}
msg: Docker API Error: Conflict. The name "api-nginx-lb" is already in use by container c4e8416e91f5. You have to delete (or rename) that container to be able to reuse that name.

FATAL: all hosts have already failed -- aborting
```

`docker ps -a` on the machine:

```
CONTAINER ID        IMAGE               COMMAND                CREATED             STATUS              PORTS               NAMES
ed29c86c5455        nginx               "nginx -g 'daemon of   11 seconds ago                                              api-nginx-lb
```

Ansible version:

```
$ ansible --version
ansible 1.9.2
  configured module search path = None
```

**Comment by robtec** [comment]
+1 on the nginx lb container

**Comment by ppawiggers** [comment]
+1 with Docker 1.7.1 and ansible 1.9.1

**Comment by ethinx** [comment]
Get the same error:

```
docker-py==1.3.1
ansible-1.9.2-1.el6.noarch

# docker version
Client version: 1.7.1
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 786b29d
OS/Arch (client): linux/amd64
Server version: 1.7.1
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 786b29d
OS/Arch (server): linux/amd64
```

**Comment by ethinx** [comment]
dig more and found this in docker's log

```
time="2015-08-11T00:24:00.365576650+08:00" level=error msg="HTTP Error" err="cannot mount volume over existing file, file exists /var/lib/docker/devicem
apper/mnt/b8c3a7865b4aa1e70d7df3ed2be8b84871e1e9714ddf0e6cf3bab6a79c4bdff9/rootfs/etc/localtime" statusCode=500
```

as I'm going to mount localtime to the container:

```
volumes:
  - /etc/localtime:/etc/localtime:ro
```

after remove the entry, container started sucessfully.

However, I should be able to mount the time.

Seems related to docker/docker-py#546

**Comment by echox** [comment]
I have exactly the same problem, even with ansible 1.9.2.

This issue seems to be a duplicate of #1337

**Comment by jonatanblue** [comment]
Same problem
ansible 1.9.2
host: Ubuntu 14.04, docker 1.8.1, docker-py 1.2.3

**Comment by Glueon** [comment]
Same problem. Appears if using volumes which overwrite files in a container.

**Comment by hrstoyanov** [comment]
Issue still in Ansible 1.9.3

**Comment by arochaf** [comment]
Does anyone have a temporary solution for this ? I'm having the same problem ...

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.
